swap/peer.go

```diff
@@ -62,27 +62,19 @@ func (sp *Peer) handleMsg(ctx context.Context, msg interface{}) error {
 
 	case *ErrorMsg:
 		return sp.handleErrorMsg(ctx, msg)
-
-	case *ConfirmMsg:
-		return sp.handleConfirmMsg(ctx, msg)
 	}
 
 	return nil
```

Comment:
Shouldn't an error be returned here? I understand at this point the message type would be something unexpected.

Reply:
In strict `go` terms you are right, the message would fall through and an "unknown message type` would be thrown.

However, because we are using `p2p/protocols`, it is actually checked before that the message passed to the handler is part of the spec:
```
354   val, ok := p.spec.NewMsg(msg.Code) 
355   if !ok { 
356     return errorf(ErrInvalidMsgCode, "%v", msg.Code) 
357   } 
358   if err := rlp.DecodeBytes(wmsg.Payload, val); err != nil { 
359     return errorf(ErrDecode, "<= %v: %v", msg, err) 
360   }
```

So it is actually not possible to receive an unknown message type. Nevertheless, I have checked other protocols we have and they still add this check. I think it makes sense because some day something may change in `p2p/protocols` and it would fall through, so to be safe I will add it as well. Thanks for raising

Reply:
Got it, thanks for the clarification.